pkg/yurtappmanager/controller/uniteddeployment/adapter/adapter_util.go

```diff
@@ -103,3 +108,29 @@ func getCurrentPartition(pods []*corev1.Pod, revision string) *int32 {
 
 	return &partition
 }
+
+func StrategicMergeByPatches(oldobj interface{}, patches *runtime.RawExtension, newPatched interface{}) error {
+	patchMap := make(map[string]interface{})
+	if err := json.Unmarshal(patches.Raw, &patchMap); err != nil {
+		klog.Errorf("Unmarshal Pool Patches error %v, Patches Raw %v", err,string(patches.Raw))
+		return  err
+	}
+
+	originalObjMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(oldobj)
+	if err != nil {
+		klog.Errorf("ToUnstructured error %v", err)
+		return  err
+	}
+
+	patchedObjMap, err := strategicpatch.StrategicMergeMapPatch(originalObjMap, patchMap, newPatched)
+	if err != nil {
+		klog.Errorf("StartegicMergeMapPatch error %v", err)
+		return err
+	}
+
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(patchedObjMap, newPatched); err != nil {
```

Comment:
Why do you need to call this?

Reply:
`StrategicMergeMapPatch`  function return a `map[string]interface{}` data.  and use `FromUnstructured` function converts an object from map[string]interface{} representation into a concrete type(Deployment or Statefulset).

and the third argument to function `StrategicMergeMapPatch` is used to determine the type, not for assignment.